packages/devtools_app/lib/src/ui/utils.dart

```diff
@@ -227,3 +228,39 @@ class ColorPair {
 
   final Color background;
 }
+
+class ThemedColorPair {
+  const ThemedColorPair({@required this.background, @required this.foreground});
+
+  factory ThemedColorPair.from(ColorPair colorPair) {
+    return ThemedColorPair(
+      foreground: ThemedColor.fromSingle(colorPair.foreground),
+      background: ThemedColor.fromSingle(colorPair.background),
+    );
+  }
+
+  final ThemedColor foreground;
+
+  final ThemedColor background;
+}
+
+/// A theme-dependent color.
+///
+/// When possible, themed colors should be specified in an extension on
+/// [ColorScheme] using the [ColorScheme.isLight] getter. However, this class
+/// may be used when access to the [BuildContext] is not available at the time
+/// the color needs to be specified.
+class ThemedColor {
+  const ThemedColor({this.light, this.dark});
+
+  factory ThemedColor.fromSingle(Color color) =>
+      ThemedColor(light: color, dark: color);
+
+  final Color light;
+
+  final Color dark;
+
+  Color colorFor(BuildContext context) {
```

Comment:
this method should take the theme or colorScheme as a parameter not a context.
That way you can lookup the theme once at the top of a build method and use it multiple places.
Theme.of calls are relatively expensive.

Reply:
done